### PR TITLE
Create some generic exceptions for adapters

### PR DESCRIPTION
**Pull request by rgazelot**
Related to https://github.com/Calendart/GoogleAdapter/pull/10

### CODE REVIEW

File: src/Exception/BadRequestException.php [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+namespace CalendArt\Exception;
+
+use Exception;
+use InvalidArgumentException;
+
+class BadRequestException extends InvalidArgumentException
+{
+    public function __construct($message = null, $code = 0, Exception $previous = null)
+    {
+        parent::__construct(null !== $message ? $message : "Bad Request", $code, $previous);
```
Comment (Taluu) [inline src/Exception/BadRequestException.php]: `$message ?: 'Bad Request'`

File: src/Exception/BackendException.php [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+namespace CalendArt\Exception;
+
+use Exception;
+
+class BackendException extends Exception
+{
+    public function __construct($message = null, $code = 0, Exception $previous = null)
+    {
+        parent::__construct(null !== $message ? $message : "Backend Error", $code, $previous);
```
Comment (Taluu) [inline src/Exception/BackendException.php]: `$message ?: 'Backend Error'`

**Comment by Taluu** [comment]
Not sure these exception are not provider related exceptions.

File: src/Exception/BadRequestException.php (hunk 1 above)
Comment (rgazelot) [inline src/Exception/BadRequestException.php]: Thanks

**Comment by rgazelot** [comment]
@Taluu Open for a review

**Comment by Taluu** [comment]
Hum, maybe adding a body would be nice. Or, how about transforming this base exception in an interface instead, even a blank one ?

**Comment by rgazelot** [comment]
@Taluu Done here too.

**Comment by Taluu** [comment]
Looks good

**Pull request merged**